aws/internal/service/sagemaker/finder/finder.go

```diff
@@ -81,6 +81,25 @@ func ImageVersionByName(conn *sagemaker.SageMaker, name string) (*sagemaker.Desc
 	return output, nil
 }
 
+// DeviceFleetByName returns the Device Fleet corresponding to the specified Device Fleet name.
+// Returns nil if no Device Fleet is found.
+func DeviceFleetByName(conn *sagemaker.SageMaker, id string) (*sagemaker.DescribeDeviceFleetOutput, error) {
+	input := &sagemaker.DescribeDeviceFleetInput{
+		DeviceFleetName: aws.String(id),
+	}
+
+	output, err := conn.DescribeDeviceFleet(input)
```

Comment:
Looking at https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_DescribeDeviceFleet.html, it looks like the API can return `ResourceNotFound` if the Device Fleet doesn't exist

Reply:
wasnt able to produce this error from tests

Reply:
moved error check to finder